### PR TITLE
[WIP] Refactor of misc1.c ins/del to cursor_modify.c

### PR DESCRIPTION
**Pull request by CaffeineViking**
I am currently trying to find related functionality in misc1.c that can be moved out, increasing readability and maintainability. Refactoring some of the code is also something I am in the process of doing. All tests have been run (functional, unit and lint). I would greatly appreciate if someone more knowledgeable could comment if this is a good idea or point out any mistakes I am making.

Please do not hesitate to comment on any mistakes I have made, this is my first try at contributing to an open source project, so there are bound to be a couple of things I have missed!
Thanks for the help!

### CODE REVIEW

File: src/nvim/cursor_modify.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#ifndef NVIM_CURSOR_MODIFY_H
+#define NVIM_CURSOR_MODIFY_H
+
+#include "nvim/vim.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "cursor_modify.h.generated.h"
+#endif
+#endif  // NVIM_CURSOR_MODIFY_H
```
Comment (Pyrohh) [inline src/nvim/cursor_modify.h]: I know it's in the style guide's example, but a space between the endif statements helps readability IMO.

File: src/nvim/cursor_modify.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#ifndef NVIM_CURSOR_MODIFY_H
+#define NVIM_CURSOR_MODIFY_H
+
+#include "nvim/vim.h"
```
Comment (Pyrohh) [inline src/nvim/cursor_modify.h]: I don't think this is needed (nvim compiles fine without it)... might be wrong though.

**Comment by Pyrohh** [comment]
Hi,

It looks like you already read [CONTRIBUTING.md](https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md) and the style guide, but just some advice:
- Commits should use imperative voice, i.e., `Moved` should be `Move`, and it could be more clear IMO. Perhaps:  ```Extract {ins,del}_* functions from misc.c to cursor_modify.c``` ? Just a suggestion
- When factoring out `(TRUE|FALSE)`, you can't simply replace all instances of them with `(true|false)`; you also need to change the types of such variables to bool, as well as the types of relevant function parameters. Due to how `(true|false)` are defined in `stdbool.h`, it compiles fine as you can see, but we don't get the type checking benefits provided for bools (at least in GCC version 5).
- Since this is a WIP, feel free to liberally edit your commit history to keep things clean. If not now, at least squash/clean up your commits before setting this PR to `RFC`. I think this is mentioned in CONTRIBUTING.md, but there's no harm in reiterating what you might have already read.

**Comment by CaffeineViking** [comment]
Thanks for the help! I will take a close look at the points you have brought forward as well as refactoring the module for readability further. 
I just wanted to ask around to see if I am progressing in the right direction, so I can continue to work on it.

**Pull request closed without merging**